recipes/eigen/all/conanfile.py

```diff
@@ -21,37 +24,42 @@ class EigenConan(ConanFile):
     }
     license = ("MPL-2.0", "LGPL-3.0-or-later")  # Taking into account the default value of MPL2_only option
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def configure(self):
         self.license = "MPL-2.0" if self.options.MPL2_only else ("MPL-2.0", "LGPL-3.0-or-later")
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def export_sources(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+            copy(self, patch["patch_file"], self.recipe_folder, self.export_sources_folder)
 
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_TESTING"] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+        tc.cache_variables["EIGEN_TEST_NOQT"] = True
+        tc.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        if not self.conf.get("tools.build:skip_test", default=True, check_type=bool):
+            cmake.build()
+            cmake.test()
```

Comment:
Is it something CCI recipes should support?

Reply:
It's not really necessary, so I can remove it.